rpcserver.go

```diff
@@ -2936,6 +2939,174 @@ func (s *rpcServer) determineAccountVersion(
 	return account.VersionInitialNoVersion, nil
 }
 
+type ActionFee struct {
+	Action  string
+	BlockHeight int32
+	TimeStamp int64
+	OutputAmount int64
+	TotalFee int64
+}
+
+// AccountActionFees returns a map from account key to an ordered list of
+// account action fees.
+func (s *rpcServer) AccountActionFees() (map[string][]ActionFee, error) {
+	// Retrieve all lightning wallet transactions.
+	ctx := context.Background()
+	transactionDetails, err := s.lndClient.GetTransactions(ctx, &lnrpc.GetTransactionsRequest{})
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving transactions: %v", err)
+	}
+
+	// Sort transactions in temporal order.
+	transactions := transactionDetails.Transactions
+	sort.Slice(transactions, func(i, j int) bool {
+		if transactions[i].BlockHeight != transactions[j].BlockHeight {
+			return transactions[i].BlockHeight < transactions[j].BlockHeight
+		}
+		return transactions[i].TimeStamp < transactions[j].TimeStamp
+	})
+
+	// A map from account key to an ordered list of account action fees.
+	accountActions := make(map[string][]ActionFee)
+
+	for _, tx := range transactions {
+		// Handle account creation action.
+		accountCreation, err := parseAccountCreationAction(tx)
+		if err != nil {
```

Comment:
can this function return an error?

Reply:
The logic which serializes account action data into labels is separate from the logic which deserializes that data. The label can included action data which the parser couldn't parse. That's where the error might arise.